threeML/plugins/SpectrumLike.py

```diff
@@ -234,6 +236,122 @@ def __init__(self, name, observation, background, verbose=True):
         # Apply the mask
         self._apply_mask_to_original_vectors()
 
+    @classmethod
```

Comment:
It looks like this shares a lot of code with the other .from_function method you wrote in this pull request.

Would it be possible to put the common code between the two somewhere else (maybe the parent class)?